pkgs/applications/version-management/git-and-tools/gh/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, buildGoModule, installShellFiles }:
+
+buildGoModule rec {
+  pname = "gh";
+  version = "0.5.3";
+
+  src = fetchFromGitHub {
+    owner = "cli";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "033y9bwdaj8735nmj22k8lrgkgimji7hyly9i4jyp11iaa7cgd7a";
+  };
+
+  modSha256 = "0ina3m2ixkkz2fws6ifwy34pmp6kn5s3j7w40alz6vmybn2smy1h";
+
+  buildFlagsArray = [
+    "-ldflags=-X github.com/cli/cli/command.Version=${version}"
+  ];
+
+  subPackages = [ "cmd/gh" ];
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    $out/bin/gh completion -s bash > gh.bash
+    $out/bin/gh completion -s fish > gh.fish
+    $out/bin/gh completion -s zsh > gh.zsh
+    installShellCompletion gh.{bash,fish,zsh}
```

Comment:
This would be better IMHO:
```bash
for shell in bash fish zsh do;
    $out/bin/gh completion -s $shell > gh.$shell
    installShellCompletion gh.$shell
done
````

Reply:
Updated.